### PR TITLE
metric json and qlog formt log for log4j2

### PR DESCRIPTION
**Pull request by lxning**
## Description

Please include a summary of the feature or issue being fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)
#1488 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the tests [UT/IT] that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced. 
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

1. json format test:

 torchserve --ncs --start --model-store model_store --models all --log-config frontend/server/src/test/resources/log4j2-json.xml

tail -f logs/ts_metrics.log
{
  "MetricName": "W-9003-BERTSeqClassification_1.0",
  "Value": "27845",
  "Unit": "ms",
  "Dimensions": [
    {
      "Name": "Level",
      "Value": "Host"
    }
  ],
  "Timestamp": "1646686889",
  "HostName": "147dda19895c.ant.amazon.com"
}
{
  "MetricName": "WorkerThreadTime",
  "Value": "90",
  "Unit": "ms",
  "Dimensions": [
    {
      "Name": "Level",
      "Value": "Host"
    }
  ],
  "HostName": "147dda19895c.ant.amazon.com"
}

2. QLogLayout
torchserve --ncs --start --model-store model_store --models all --log-config frontend/server/src/test/resources/log4j2-qlog.xml

tail -f logs/ts_metrics.log
HostName=147dda19895c.ant.amazon.com
StartTime=1646686978
Program=MXNetModelServer
Metrics=MemoryUsed=5790.98046875 Megabytes Level|Host
EOE
HostName=147dda19895c.ant.amazon.com
StartTime=1646686978
Program=MXNetModelServer
Metrics=MemoryUtilization=46.2 Percent Level|Host
EOE

## Checklist:

- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: frontend/server/src/test/resources/log4j2-json.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (msaroufim) [inline frontend/server/src/test/resources/log4j2-json.xml:1]: Could you add comments on each of the xml files on how you generated them?

Were there any existing xml files that now need to be deleted

Where are these xml files used?

Generally feels like we need to update internals guide with how metrics work

File: frontend/server/src/main/java/org/pytorch/serve/util/logging/QLogLayout.java [hunk 2]
```diff
@@ -19,6 +20,11 @@ public QLogLayout() {
         super(null, null, null);
     }
 
+    @PluginFactory
```
Comment (msaroufim) [inline frontend/server/src/main/java/org/pytorch/serve/util/logging/QLogLayout.java:23]: What does this line do exactly?

**Review by msaroufim — CHANGES_REQUESTED** [review]
Left some high level feedback, could you please also add the broken vs expected behavior logs in the description and explain at a high level how this PR works - why do we have 2 new plugins and how do they use the xml files and how were those xml files generated and how will we maintain them.

File: frontend/server/src/main/java/org/pytorch/serve/util/logging/JSONPatternLayout.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public JSONPatternLayout() {
+        super(null, null, null);
+    }
+
+    @PluginFactory
+    public static JSONPatternLayout createLayout() {
+        return new JSONPatternLayout();
+    }
+
+    @Override
```
Comment (maaquib) [inline frontend/server/src/main/java/org/pytorch/serve/util/logging/JSONPatternLayout.java:28]: Add javadoc similar to `QLogLayout` for the expected log format

**Review by maaquib — CHANGES_REQUESTED** [review]

**Comment by maaquib** [comment]
Some formatting build failures in `verifyJava` needs to be fixed as well for CI

**Comment by lxning** [comment]
@msaroufim 

1. #1488 describes the expected behavior vs current behavior
2. the above test results show how to use and result.
3. please refer [doc](https://logging.apache.org/log4j/2.x/manual/configuration.html) for writing log4j2.xml

**Review by maaquib — APPROVED** [review]

**Review by msaroufim — APPROVED** [review]

**Pull request merged**